app/javascript/src/utils/compiler/variables.js

```diff
@@ -164,8 +164,8 @@ export function getVariables(joinedItems) {
   const literalGlobalVariables = matchAllOutsideRanges(stringRanges, joinedItems, globalVariablesRegex).map((match) => match[0])
   const literalPlayerVariables = getLiteralPlayerVariables(joinedItems, stringRanges)
 
-  const playerVariables = [...new Set([...literalPlayerVariables, ...playerVariablesFromActions])]
-  const globalVariables = [...new Set([...literalGlobalVariables, ...globalVariablesFromActions])]
+  const playerVariables = [...new Set([...literalPlayerVariables, ...playerVariablesFromActions])].filter(v => v)
+  const globalVariables = [...new Set([...literalGlobalVariables, ...globalVariablesFromActions])].filter(v => v)
```

Comment:
Typecheck issue?

Reply:
No, it actually returned undefined. Not sure why it didn't before. I didn't debug very much

Reply:
It wasn't a project, the tests were failing. This one specifically: https://github.com/Mitcheljager/workshop.codes/blob/4e67b79c04b8a1496bfaf7a0f15973bc9534155d/app/javascript/test/utils/compiler/variables.test.js#L316-L328

Reply:
That's interesting. I think it may have been a bug with Jest's compare logic (or maybe we were using the wrong function?).

![clearly not the same output](https://github.com/Mitcheljager/workshop.codes/assets/6181929/e698243a-08d7-4caa-a321-e833cfe8d0b0)
(Using Jest here - these objects are _not_ the same)

Vite correctly compares the objects how we want it to, so that's why this issue appeared.
I will fix the underlying issue on another PR.